include/grpcpp/impl/codegen/sync.h

```diff
@@ -19,8 +19,15 @@
 #ifndef GRPCPP_IMPL_CODEGEN_SYNC_H
 #define GRPCPP_IMPL_CODEGEN_SYNC_H
 
-#include <grpc/impl/codegen/log.h>
 #include <grpc/impl/codegen/port_platform.h>
+
+#ifndef GPR_WINDOWS
```

Comment:
Hmm... Do we want this kind of conditional outside of port_platform.h?

@nicolasnoble should weigh in on this.

Reply:
Here, we have to include `<pthread>` directly without relying on `GPR_SYNC_POSIX` to make sure  the space is large enough wherever `pthread_mutex_t` can possibly be use in place of `gpr_mu`.